### PR TITLE
Added CONTRIBUTING.md as requested in an issue.

### PR DESCRIPTION
**Pull request by theballmarcus**
You asked for a CONTRIBUTING.md - here it is. It should be fairly simple to add desired topics. :)

### CODE REVIEW

File: CONTRIBUTING.md
Comment (capnkenny) [inline CONTRIBUTING.md:26]: We will need a little more detail here on how contributors can get started 😅 apologies for not having the issue fully written out at the time you picked this up.

More specifically, we want to cover the following:

- How to Get in Touch
-- If they want to ask about a ticket, make a suggestion, show off how they're using NovelRT, if they need assistance with the project or even just chat, they can reach us at Discord 😄 
-- They can also open an issue or start a Github Discussion, too
-- Also, if they have additional (relevant) input on currently open issues, they can feel free to comment as well

- Looking for an Issue
-- Call out here that we do use `help wanted` and `good first issue` labels, but any issue that is unassigned can be picked up.
-- If the issue you're looking at is assigned but hasn't been worked on in a considerable amount of time, reach out to the maintainers about reassignment.

- Contributing to Development (once an Issue has been picked up)
-- This area should link back on how to build NovelRT (from the README)
-- It should also mention our Code Style - we use an automatic formatter `clang-format` to adhere to a codestyle similar to Microsoft's (https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options?view=vs-2017)
-- Also should mention that if they have trouble adhering to the code style, our CI process will produce a patch for them when needed so that they can apply it and make their code adherent.

- Creating a Pull Request
-- This should mention that PRs need to be directed to the `main` branch, but should start from a fork on a different branch.
-- You can specifically call out if they are a first-timer, they can look to the basic links that you provided here.
-- PRs should be tagged with an issue they resolve - PRs without issue need to be briefly discussed with the maintainers first before they will be accepted.
-- Once a PR has been submitted, one or more of the maintainers will review, comment, and/or merge based on if the pull request meets criteria or not.
-- It should be specifically called out that any pull requests created must have their creator agree to our Contributor License Agreement (CLA) before the pull request will be allowed to be merged.

If you'd like a better idea of how the flow of the guide should be, feel free to look here as a reference: https://reactjs.org/docs/how-to-contribute.html

Also, for this piece, you can tag me in comments or reach out to me however if you need additional follow-up or info to help you complete this!

File: CONTRIBUTING.md
Comment (capnkenny) [inline CONTRIBUTING.md:3]:
```suggestion
NovelRT is a community-driven project. Contribution is always welcome, and much appreciated.
```

Just a suggestion here as always appears to be mistyped

File: CONTRIBUTING.md
Comment (capnkenny) [inline CONTRIBUTING.md:6]:
```suggestion
First, please take a moment to review our [code of conduct](CODE_OF_CONDUCT.md). We expect that all project participants will adhere to it, and any actions taken that go against the Code of Conduct will not be tolerated.
```

Good first line here! Just added a suggestion as we want to convey that actions against the CoC have zero tolerance in our eyes.

File: CONTRIBUTING.md
Comment (capnkenny) [inline CONTRIBUTING.md:9]:
```suggestion
These guidelines are an attempt at better addressing the incoming issues and pull requests. 
Please read them closely, and feel free to ask any questions if needed.
```

Just a suggestion here as the softer language makes it sound generally more inviting 🙂

File: CONTRIBUTING.md
Comment (capnkenny) [inline CONTRIBUTING.md:13]:
```suggestion
## Getting started

### Submitting Pull Requests
```

Suggestion: Reverse these? It seems to read easier this way (in the context of the flow of the document).

File: CONTRIBUTING.md
Comment (capnkenny) [inline CONTRIBUTING.md:28]:
```suggestion
Finally, thank you to [all of our current and future contributors!](https://github.com/novelrt/NovelRT/graphs/contributors)
```

One last suggestion here - just tweaked it a bit and corrected 1 typo (I like the premise here though)

**Review by capnkenny — CHANGES_REQUESTED** [review]
Hey!
So far, it looks like you're off to a good start! 😄  

There _are_ some tweaks/changes for you to consider, and we'd like your help with detailing out more of the document. Let me know if you have any questions on the feedback submitted and one of us (probably me lol) will follow up as soon as we can.

**Comment by RubyNova** [comment]
hey there, just wanted to flag I think something is wrong with out Ubuntu CI - don't worry about it, we're handling it internally. :)

**Comment by theballmarcus** [comment]
I added quite a lot more to the contribuing file. Let me know what you think! :)

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:4]: Unsure if this line is really neccessary, as the statement right before this kind of states this, I feel?

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:6]: I personally think this reads better:

```suggestion
First, please take a moment to review our [Code of Conduct](CODE_OF_CONDUCT.md). We expect that all project participants will adhere to it, and violations will not be tolerated.

You can forward violations or other concerns to admin@novelrt.dev or via a direct message to one of the maintainers on our [Discord community server](https://discord.novelrt.dev).
```

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:9]: @capnkenny gonna want you to check my suggestion here, I'm not 100% sure on my suggestion here but I do think the original could be improved.

```suggestion
These guidelines are an outline of the contribution process. Please read them closely, and feel free to ask any questions if needed.
```

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:16]: I think we have two options here.

You can either do this:
```suggestion
You might know the basics of [contributing on GitHub](https://help.github.com/articles/using-pull-requests) and have a fork [properly set up](https://github.com/ohmyzsh/ohmyzsh/wiki/Contribution-Technical-Practices).
```

Or you can do this:
```suggestion
Before submitting a pull request (PR), please ensure you are familiar with the following: 
- [Contributing on GitHub](https://help.github.com/articles/using-pull-requests)
- [Properly setting up a fork](https://github.com/ohmyzsh/ohmyzsh/wiki/Contribution-Technical-Practices).
```

I personally prefer the latter, but I'm open to discuss it.

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:21]: I think the message would come across clearer like so:

```suggestion
When creating a PR, please provide an outline of what you fixed or added in the PR body with relevant GitHub keywords and the issue number(s) in question. You can find more information on this [here](https://help.github.com/articles/closing-issues- via-commit-messages/).

Please also ensure that if you add new functionality, and it is testable, that you also add the relevant unit tests, if none exist.

While any feedback that is given from the maintainer team or one of our other contributors should be assumed to be in good faith and for the betterment of the NovelRT project and its sister repositories, please forward any concerns to the relevant places outlined at the beginning of this document.
```

It's more wordy, but I feel this covers the message better. :)
cc @capnkenny thoughts here?

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:58]: Using our official URL to make life easier. :)

```suggestion
### You can always reach us on [Discord](https://discord.novelrt.dev) if you want to ask about a ticket, make a suggestion, show off how you're using NovelRT, if you need assistance with the project or even just chat. :) 
```

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:29]: Just adding some missing context here. :)

```suggestion
We use the ```clang-format``` code formatter configured to the Microsoft code formatting style, and will be using ```clang-tidy``` in the future to enforce conventions such as casing. For now, though, please ensure that your code follows the common C# styles, as used in places such as the [.NET Runtime](https://github.com/dotnet/runtime) and many other projects.
```

File: CONTRIBUTING.md
Comment (RubyNova) [inline CONTRIBUTING.md:30]: Fixing a spelling error and tweaking the wording a little with this suggestion.

```suggestion
If you find it hard to adhere to the code style, our CI process will produce a patch for you when needed so that you can apply it and make your code consistent with the rest of the project.
```

**Review by RubyNova — CHANGES_REQUESTED** [review]
Looking pretty solid for our CONTRIBUTING.md, just put some of my own thoughts and corrections in here while reading through it. I imagine this will either be the penultimate or ultimate last set of requested changes you''ll have to make. :D 

@capnkenny can you give these review points the once over as well?

File: CONTRIBUTING.md
Comment (capnkenny) [reply CONTRIBUTING.md:9]: I think it summarizes the phrasing well here @RubyNova

File: CONTRIBUTING.md
Comment (capnkenny) [reply CONTRIBUTING.md:16]: I'd agree to the latter as well

File: CONTRIBUTING.md
Comment (capnkenny) [reply CONTRIBUTING.md:21]: Its more concise and direct in this version, @RubyNova

**Comment by RubyNova** [comment]
Hey @theballmarcus , how is this PR going?

**Comment by theballmarcus** [comment]
Hey - sorry for it taking a long time, something happened that made me busy. I will get it done soon though. :)

**Comment by RubyNova** [comment]
No worries - was just checking in on you! 😊

**Comment by RubyNova** [comment]
Assuming the PR is dead. Closing.

**Pull request closed without merging**
